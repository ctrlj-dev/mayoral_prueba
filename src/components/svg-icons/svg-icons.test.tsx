import { render } from '@testing-library/react';
import { CrossIcon, LineIcon, SearchIcon } from './svg-icons.component';

describe('SVG Icons specs', () => {
    test('Test for Icons to check if are render properly', () => {
        //Act
        const screen = render(
            <>
                <div data-testid="searchIcon">
                    <SearchIcon
                        id={"searchIcon"}
                        className={'searchIcon'}
                        width={20}
                        height={20}
                    />
                </div>
                <div data-testid="lineIcon">
                    <LineIcon
                        id={"lineIcon"}
                        className={'lineIcon'}
                        width={20}
                        height={20}
                    />
                </div>
                <div data-testid="crossIcon">
                    <CrossIcon
                        id={"crossIcon"}
                        className={'crossIcon'}
                        width={20}
                        height={20}
                    />
                </div>
            </>
        )

        const searchIcon = screen.getByTestId('searchIcon').firstChild.nodeName;
        const lineIcon = screen.getByTestId('lineIcon').firstChild.nodeName;
        const crossIcon = screen.getByTestId('crossIcon').firstChild.nodeName;
        //Assert
        expect(searchIcon).not.toBeNull();
        expect(searchIcon).toEqual('svg')
        expect(screen.asFragment()).toMatchSnapshot();

        expect(lineIcon).not.toBeNull();
        expect(lineIcon).toEqual('svg')
        expect(screen.asFragment()).toMatchSnapshot();

        expect(crossIcon).not.toBeNull();
        expect(crossIcon).toEqual('svg')
        expect(screen.asFragment()).toMatchSnapshot();

    })
})
