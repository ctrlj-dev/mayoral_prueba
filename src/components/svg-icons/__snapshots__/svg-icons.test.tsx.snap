// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG Icons specs Test for Icons to check if are render properly 1`] = `
<DocumentFragment>
  <div
    data-testid="searchIcon"
  >
    <svg
      class="searchIcon"
      height="20"
      id="searchIcon"
      version="1.1"
      viewBox="0 0 487.95 487.95"
      width="20"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M481.8,453l-140-140.1c27.6-33.1,44.2-75.4,44.2-121.6C386,85.9,299.5,0.2,193.1,0.2S0,86,0,191.4s86.5,191.1,192.9,191.1 c45.2,0,86.8-15.5,119.8-41.4l140.5,140.5c8.2,8.2,20.4,8.2,28.6,0C490,473.4,490,461.2,481.8,453z M41,191.4 c0-82.8,68.2-150.1,151.9-150.1s151.9,67.3,151.9,150.1s-68.2,150.1-151.9,150.1S41,274.1,41,191.4z"
      />
    </svg>
  </div>
  <div
    data-testid="lineIcon"
  >
    <svg
      class="lineIcon"
      height="20"
      id="lineIcon"
      viewBox="0 0 45 45"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="10"
        width="45"
        y="17.5"
      />
    </svg>
  </div>
  <div
    data-testid="crossIcon"
  >
    <svg
      class="crossIcon"
      height="20"
      id="crossIcon"
      viewBox="0 0 45 45"
      width="20"
    >
      <rect
        height="10"
        width="45"
        y="17.5"
      />
      <rect
        height="10"
        transform="translate(0 45) rotate(-90)"
        width="45"
        y="17.5"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`SVG Icons specs Test for Icons to check if are render properly 2`] = `
<DocumentFragment>
  <div
    data-testid="searchIcon"
  >
    <svg
      class="searchIcon"
      height="20"
      id="searchIcon"
      version="1.1"
      viewBox="0 0 487.95 487.95"
      width="20"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M481.8,453l-140-140.1c27.6-33.1,44.2-75.4,44.2-121.6C386,85.9,299.5,0.2,193.1,0.2S0,86,0,191.4s86.5,191.1,192.9,191.1 c45.2,0,86.8-15.5,119.8-41.4l140.5,140.5c8.2,8.2,20.4,8.2,28.6,0C490,473.4,490,461.2,481.8,453z M41,191.4 c0-82.8,68.2-150.1,151.9-150.1s151.9,67.3,151.9,150.1s-68.2,150.1-151.9,150.1S41,274.1,41,191.4z"
      />
    </svg>
  </div>
  <div
    data-testid="lineIcon"
  >
    <svg
      class="lineIcon"
      height="20"
      id="lineIcon"
      viewBox="0 0 45 45"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="10"
        width="45"
        y="17.5"
      />
    </svg>
  </div>
  <div
    data-testid="crossIcon"
  >
    <svg
      class="crossIcon"
      height="20"
      id="crossIcon"
      viewBox="0 0 45 45"
      width="20"
    >
      <rect
        height="10"
        width="45"
        y="17.5"
      />
      <rect
        height="10"
        transform="translate(0 45) rotate(-90)"
        width="45"
        y="17.5"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`SVG Icons specs Test for Icons to check if are render properly 3`] = `
<DocumentFragment>
  <div
    data-testid="searchIcon"
  >
    <svg
      class="searchIcon"
      height="20"
      id="searchIcon"
      version="1.1"
      viewBox="0 0 487.95 487.95"
      width="20"
      x="0px"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M481.8,453l-140-140.1c27.6-33.1,44.2-75.4,44.2-121.6C386,85.9,299.5,0.2,193.1,0.2S0,86,0,191.4s86.5,191.1,192.9,191.1 c45.2,0,86.8-15.5,119.8-41.4l140.5,140.5c8.2,8.2,20.4,8.2,28.6,0C490,473.4,490,461.2,481.8,453z M41,191.4 c0-82.8,68.2-150.1,151.9-150.1s151.9,67.3,151.9,150.1s-68.2,150.1-151.9,150.1S41,274.1,41,191.4z"
      />
    </svg>
  </div>
  <div
    data-testid="lineIcon"
  >
    <svg
      class="lineIcon"
      height="20"
      id="lineIcon"
      viewBox="0 0 45 45"
      width="20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <rect
        height="10"
        width="45"
        y="17.5"
      />
    </svg>
  </div>
  <div
    data-testid="crossIcon"
  >
    <svg
      class="crossIcon"
      height="20"
      id="crossIcon"
      viewBox="0 0 45 45"
      width="20"
    >
      <rect
        height="10"
        width="45"
        y="17.5"
      />
      <rect
        height="10"
        transform="translate(0 45) rotate(-90)"
        width="45"
        y="17.5"
      />
    </svg>
  </div>
</DocumentFragment>
`;
